src/core/tests/tensor.cpp

```diff
@@ -88,5 +94,251 @@ TEST(tensor, get_byte_size_u6_not_even_div_by_storage_unit) {
+    std::tuple<std::filesystem::path, ov::float8_e5m2>,
+    std::tuple<std::filesystem::path, ov::float4_e2m1>,
+    std::tuple<std::filesystem::path, ov::float8_e8m0>>;
+
+INSTANTIATE_TYPED_TEST_SUITE_P(OffloadTensorTest, ParametredOffloadTensorTest, TypesToTest);
+
+TEST(OffloadTensorTest, string_tensor_throws) {
+    ov::Tensor str_tensor(ov::element::string, ov::Shape{1});
+    auto file_name = ov::test::utils::generateTestFilePrefix();
+    {
+        EXPECT_THROW(save_tensor_data(str_tensor, file_name), ov::Exception);
+        ASSERT_FALSE(std::filesystem::exists(file_name));
+    }
+    {
+        std::ofstream fout(file_name, std::ios::binary);
+        fout << "Hello, world!";
+        fout.close();
+        ASSERT_TRUE(std::filesystem::exists(file_name));
+        EXPECT_THROW(read_tensor_data(file_name, str_tensor), ov::Exception);
+        std::filesystem::remove(file_name);
+    }
+    {
+        EXPECT_THROW(std::ignore = ov::create_mmaped_tensor(str_tensor, file_name), ov::Exception);
+        ASSERT_FALSE(std::filesystem::exists(file_name));
+    }
+}
+
+class FunctionalOffloadTensorTest : public ::testing::Test {
+public:
+    void SetUp() override {
+        initial_tensor = Tensor(ov_type, shape.get_shape());
+        std::vector<float> init_values(initial_tensor.get_size());
+        ov::test::utils::fill_data_random(init_values.data(), initial_tensor.get_size(), 10, 0, 100);
+        std::memcpy(initial_tensor.data(), init_values.data(), initial_tensor.get_byte_size());
+
+        file_name = ov::test::utils::generateTestFilePrefix();
+    }
+
+    void remove_file() {
+        if (std::filesystem::exists(file_name))
+            std::filesystem::remove(file_name);
+    }
+
+    ov::element::Type ov_type{ov::element::f32};
+    ov::PartialShape shape{1, 2, 3, 4};
+    ov::Tensor initial_tensor;
+    std::string file_name;
+};
+
+TEST_F(FunctionalOffloadTensorTest, read_with_offset) {
+    auto data_size = initial_tensor.get_byte_size();
+    {
+        float dummy = 0;
+        std::ofstream fout(file_name, std::ios::binary);
+        fout.write(reinterpret_cast<char*>(&dummy), sizeof(float));
+        fout.write(reinterpret_cast<char*>(initial_tensor.data()), data_size);
+    }
+    ASSERT_TRUE(std::filesystem::exists(file_name));
+
+    {
+        ov::Tensor tensor;
+        EXPECT_NO_THROW(tensor = read_tensor_data(file_name, ov_type, shape, sizeof(float), true));
+        EXPECT_EQ(0, memcmp(tensor.data(), initial_tensor.data(), data_size));
+    }
+    remove_file();
+}
+
+TEST_F(FunctionalOffloadTensorTest, read_without_mmap) {
+    auto data_size = initial_tensor.get_byte_size();
+    {
+        std::ofstream fout(file_name, std::ios::binary);
+        fout.write(reinterpret_cast<char*>(initial_tensor.data()), data_size);
+    }
+    ASSERT_TRUE(std::filesystem::exists(file_name));
+
+    {
+        ov::Tensor tensor;
+        EXPECT_NO_THROW(tensor = read_tensor_data(file_name, ov_type, shape, 0, false));
```

Comment:
Are there test where offset is outside of tensor shape or file size?

Reply:
added tests